### PR TITLE
add compatibility with manifest V3

### LINKED ISSUE #28: MV3 Support?
It will support MV3?
thanks.

### PR DESCRIPTION
**Pull request by lmichelin**
This PR adds the compatibility with the manifest V3 without breaking the compatibility with the manifest V2.

#### Checklist

- [x] I ran the linting and formatting tools (ESLint and Prettier)
- [ ] I added addittional tests if adding new features

Closes #28

### CODE REVIEW

File: src/middleware/wer-middleware.raw.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@
 /*  This will be converted into a lodash templ., any  */
 /*  external argument must be provided using it       */
 /* -------------------------------------------------- */
-(function(window) {
+(function() {
```
Comment (sherifmayika) [inline src/middleware/wer-middleware.raw.ts:8]: I locally compiled and tested with Manifest v3. I still get error **Uncaught ReferenceError: window is not defined**
at
`var injectionContext = this || window || {`

Comment (sherifmayika) [reply src/middleware/wer-middleware.raw.ts:8]: commented `var injectionContext = this || /*window ||*/ {` then "window" error is gone in MV 3. not tested with Mv2.  But now get `Uncaught TypeError: Cannot read properties of null (reading 'extension')`  `var extension = browser.extension,`

Comment (lmichelin) [reply src/middleware/wer-middleware.raw.ts:8]: Could you explain how to reproduce the bug? (Browser version, manifest content, steps to follow...). Thanks!

Comment (sherifmayika) [reply src/middleware/wer-middleware.raw.ts:8]: Chromium
Version 107.0.5304.110 (Official Build) Arch Linux (64-bit)


Manifest 
`{
  "manifest_version": 3,
  "name": "__MSG_extName__",
  "homepage_url": "http://localhost:8080/",
  "description": "A Vue Browser Extension",
  "default_locale": "en",
  "permissions": [
    "activeTab"
  ],
  "icons": {
    "16": "icons/16.png",
    "48": "icons/48.png",
    "128": "icons/128.png"
  },
  "background": {
    "service_worker": "js/background.js",
    "type": "module"
  },
  "content_security_policy": {
    "extension_pages": "script-src 'self'; object-src 'self'; script-src-elem 'self' 'unsafe-inline'; connect-src https://* data: blob: filesystem:"
  },
  "action": {
    "default_popup": "popup.html",
    "default_title": "__MSG_extName__",
    "default_icon": {
      "19": "icons/19.png",
      "38": "icons/38.png"
    }
  },
  "version": "0.1.0"
}
`
no content script
Background

`
chrome.action.onClicked.addListener(() => {
  console.log("browser action clicked");
  // openApp("popup.html");
  chrome.tabs.create({
    url: chrome.runtime.getURL("popup.html"),
  });
});

`

Comment (sherifmayika) [reply src/middleware/wer-middleware.raw.ts:8]: @lmichelin 
`-/* eslint:disable */
+/* eslint-disable */`

`-(function(window) {

+(function() {
+try {window} catch {var window: any;};`

this solves the "window is not defined" error


to remove 'extension'is undefined error 

`- const { extension, runtime, tabs } = browser;
  const manifest = runtime.getManifest();

  // =============================== Helper functions ======================================= //
  
  
  
  + const { runtime, tabs } = browser;
  const manifest = runtime.getManifest();

  // =============================== Helper functions ======================================= //`
  
  
[https://github.com/voyage-finance/webpack-ext-reloader/pull/1/commits/3c96c5c283c4835408c8eacce2da84e9f58f88cd](https://github.com/voyage-finance/webpack-ext-reloader/pull/1/commits/3c96c5c283c4835408c8eacce2da84e9f58f88cd)

to resolve "runtime" is undefined error 
 
 I replaced `browser.runtime` with `chrome.runtime` and `browser.tabs` with `chrome.tabs` in 
  
`node_modules/webpack-ext-reloader/dist/webp`ack-ext-reloader.js`



now everything working fine for me.


`browser.runtime` comes from 'webextension-polyfill'


[https://github.com/mozilla/webextension-polyfill/issues/329](https://github.com/mozilla/webextension-polyfill/issues/329) has clue how to make changes in  `wer-middleware.raw.ts`

**Pull request merged**